src/generate-query.ts

```diff
@@ -707,32 +709,24 @@ function getListSizeDirectiveDefaultValue(
 
   const requireOneSlicingArg = getRequireOneSlicingArgument(listSizeDirective)
   if (
-    !requireOneSlicingArg ||
-    (requireOneSlicingArg.value as BooleanValueNode).value === true
+    requireOneSlicingArg &&
+    (requireOneSlicingArg.value as BooleanValueNode).value === false
   ) {
-    slicingArgumentsPaths = slicingArgumentsPaths.slice(0, 1)
+    throw new Error(
+      'getListSizeDirectiveDefaultValue does not handle cases where requiredOneSlicingArg is set to false'
+    )
   }
-  const slicingArgumentsTokenizedPaths = slicingArgumentsPaths.map((value) =>
-    value.split('.')
-  )
+  const slicingArgumentsTokenizedPath = slicingArgumentsPaths[0].split('.')
 
-  if (
-    slicingArgumentsPaths.length === 1 &&
-    slicingArgumentsTokenizedPaths[0].length === 1
-  )
+  if (slicingArgumentsTokenizedPath.length === 1)
     return getDefaultArgValue(schema, config, typeNode)
 
-  return slicingArgumentsTokenizedPaths.reduce((res, tokenizedPath) => {
```

Comment:
I think what you had here is pretty sensible (i.e. slicing `slicingArgumentsTokenizedPaths` when `requireOneSlicingArg` is set to true, otherwise iterating over all the paths using a `reduce`).

What you could do to add multiple default values (if `requireOneSlicingArg` is set to true`) is, 1) pass in the `variableValues` and 2) instead of doing a `reduce()`, you can just do a `forEach()` and in the call back, add the value into `variableValues`.

Reply:
Instead of immediately adding new entries to the `variableValues`, I think we should consider checking if 1) the key already exists in which case we should throw an error (because our code is supposed to ignore variables that have already been provided a value) and 2) attempt a merge when the type of the variable is an input object